test/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.test.cpp

```diff
@@ -486,6 +496,380 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Maneuver, ToTabulatedDynamics)
+{
+    const Shared<const Frame> inertialFrameSPtr = Frame::GCRF();
+    const Shared<const LocalOrbitalFrameFactory> localOrbitalFrameFactorySPtr =
+        LocalOrbitalFrameFactory::TNW(inertialFrameSPtr);
+
+    {
+        const Maneuver::MeanDirectionAndMaximumAngularOffset meanDirectionAndMaximumAngularOffset =
+            defaultManeuver_.calculateMeanThrustDirectionAndMaximumAngularOffset(localOrbitalFrameFactorySPtr);
+
+        const LocalOrbitalFrameDirection meanDirection = meanDirectionAndMaximumAngularOffset.first;
+        EXPECT_TRUE(meanDirection.isDefined());
+        EXPECT_EQ(meanDirection.getLocalOrbitalFrameFactory(), localOrbitalFrameFactorySPtr);
+
+        const Angle maximumAngularOffset = meanDirectionAndMaximumAngularOffset.second;
+        EXPECT_TRUE(maximumAngularOffset.isDefined());
+    }
+
+    {
+        EXPECT_THROW(
+            {
+                try
+                {
+                    defaultManeuver_.calculateMeanThrustDirectionAndMaximumAngularOffset(nullptr);
+                }
+                catch (const ostk::core::error::runtime::Undefined& e)
+                {
+                    EXPECT_EQ("{Local Orbital Frame Factory} is undefined.", e.getMessage());
+                    throw;
+                }
+            },
+            ostk::core::error::runtime::Undefined
+        );
+    }
+
+    {
+        EXPECT_THROW(
+            {
+                try
+                {
+                    defaultManeuver_.calculateMeanThrustDirectionAndMaximumAngularOffset(
+                        LocalOrbitalFrameFactory::Undefined()
+                    );
+                }
+                catch (const ostk::core::error::runtime::Undefined& e)
+                {
+                    EXPECT_EQ("{Local Orbital Frame Factory} is undefined.", e.getMessage());
+                    throw;
+                }
+            },
+            ostk::core::error::runtime::Undefined
+        );
+    }
+
+    {
+        const Array<State> statesWithZeroThrustAcceleration = {
+            stateGenerator(Instant::J2000() + Duration::Seconds(1.0), {0.0, 0.0, 0.0}, -1.0e-5),
+            stateGenerator(Instant::J2000() + Duration::Seconds(5.0), {0.0, 0.0, 0.0}, -1.0e-5),
+        };
+        const Maneuver maneuverWithZeroThrustAcceleration = {statesWithZeroThrustAcceleration};
+
+        EXPECT_THROW(
+            maneuverWithZeroThrustAcceleration.calculateMeanThrustDirectionAndMaximumAngularOffset(
+                localOrbitalFrameFactorySPtr
+            ),
+            ostk::core::error::RuntimeError
+        );
+    }
+
+    {
+        // Acceleration direction is constant in LoF frame ({2, -1, 3}, in TNW), with different magnitudes at each
+        // state: 0.8 and 1.5.
+        const Array<State> originalStates = {
+            stateGenerator(
+                Instant::J2000() + Duration::Seconds(0.0),
+                {0.8 * 1.0, 0.8 * 2.0, 0.8 * 3.0},  // 0.8 * {2, -1, 3} in TNW
```

Comment:
hmmm `{2, -1, 3}` is not a unit vector - Can you make it a unit vector here so that the magnitudes listed in the unit test are truly the magnitudes of the acceleration vector?